pkg/controller/direct/directbase/directbase_controller.go

```diff
@@ -102,6 +106,52 @@ func add(mgr manager.Manager, r *DirectReconciler, reconcilePredicate predicate.
 	obj := &unstructured.Unstructured{}
 	obj.SetGroupVersionKind(r.gvk)
 
+	// For GVKs that model resources with mutable but unreadable fields (including sensitive fields),
+	// we add an event handler to cache the most recent old object update
+	// TODO acpana: better GVK abstraction for registration
+	if r.gvk.Kind == "LoggingLogMetric" {
+		log.FromContext(context.Background()).Info("adding update handler with full object fetch for LoggingLogMetric")
+
+		// Use PartialObjectMetadata in informer (because builder.OnlyMetadata is used)
+		metaObj := &metav1.PartialObjectMetadata{}
+		metaObj.SetGroupVersionKind(r.gvk)
+
+		informer, err := mgr.GetCache().GetInformer(context.Background(), metaObj)
+		if err != nil {
+			return fmt.Errorf("error getting informer for %s: %w", r.gvk, err)
+		}
+		_, err = informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+			UpdateFunc: func(oldObjMeta, newObjMeta interface{}) {
+				if oldObjMeta == nil || newObjMeta == nil {
+					return
+				}
+				oldMeta, ok := oldObjMeta.(*metav1.PartialObjectMetadata)
+				if !ok {
+					log.FromContext(context.Background()).Error(fmt.Errorf("expected oldObjMeta to be of type *metav1.PartialObjectMetadata)"), "error converting oldObjMeta in UpdateEvent handler")
+					return
+				}
+				key := types.NamespacedName{
+					Name:      oldMeta.GetName(),
+					Namespace: oldMeta.GetNamespace(),
+				}
+
+				// Fetch the full object manually using client.Client
+				fullObj := &unstructured.Unstructured{}
+				fullObj.SetGroupVersionKind(r.gvk)
+
+				if err := mgr.GetClient().Get(context.Background(), key, fullObj); err != nil {
```

Comment:
I understand we are using the old obj meta to fetch the object. But isn't the object we're getting already updated?